android: Implement Adreno Turbo setting for eligible GPUs

kleidis opened the pull request:
I can't test this myself because I don't own a Qualcomm device anymore, but *it should work*

- The toggle is under Disk shader cache in the graphics settings 

Credits to @ishan09811

Ishan09811 committed:
Implement Force Maximum GPU Clocks option  (#42)

* bonk

* some fixes

* fix syntax

* Implement force maximum gpu clock option

* fix ``enableAdrenoTurboMode`` boolean

* make force maximum gpu clocks option not editable if device doesn't support

* fix

* one more fix

* Make ``FORCE_MAX_GPU_CLOCKS`` not runtime editable for now

* some fixes

* properly get settings items

* bonk

* show correct dialog when force max gpu clocks is not supported

* typo

* fix adrenotools_set_turbo

* fix one more typo

* Fix clang-format

---------

Co-authored-by: Gamer64 <76565986+Gamer64ytb@users.noreply.github.com>

Commit:
Fix lime-format and typos

OpenSauce04 commented on src/android/app/src/main/res/values/strings.xml:
```diff
@@ -263,6 +263,10 @@
+    <string name="force_max_gpu_clocks">Force Maximum GPU Clocks (Adreno Only)</string>
+    <string name="force_max_gpu_clocks_description">Forces the GPU to run at its maximum possible clock speed (May cause excessive heating and power usage)</string>
+    <string name="force_max_gpu_clocks_desc_unsupported">Your device does not support forcing maximum GPU clocks</string>
+    <string name="force_max_gpu_clocks_unsupported">Forcing maximum GPU clocks not supported</string>
```
Please replace all instances of "Clocks" with "Clock Speed". It is not correct terminology.

This includes internal variable names and string keys.

OpenSauce04 commented on src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/SwitchSettingViewHolder.kt:
```diff
@@ -33,10 +37,20 @@ class SwitchSettingViewHolder(val binding: ListItemSettingSwitchBinding, adapter
+        isForceMaxGpuClocksClickable()
+        } else { 
+            setting.isEditable
+        }
```
This can be accomplished in a single line by simply using an `&&` operator. No if statement necessary.

OpenSauce04 commented on src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/SwitchSettingViewHolder.kt:
```diff
@@ -45,18 +59,36 @@ class SwitchSettingViewHolder(val binding: ListItemSettingSwitchBinding, adapter
+        } else {
+            true
+        }
+    }
```
Having the logic for handling one specific setting sprinkled all over generic switch setting code is not acceptable.
If this is absolutely necessary, a generic implementation should be made

OpenSauce04 commented on src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/SettingsAdapter.kt:
```diff
@@ -451,6 +451,13 @@ class SettingsAdapter(
+            R.string.force_max_gpu_clocks_unsupported,
+            R.string.force_max_gpu_clocks_desc_unsupported
+        ).show((fragmentView as SettingsFragment).childFragmentManager, MessageDialogFragment.TAG)
+    }
```
What I said about `isForceMaxGpuClocksClickable` applies here as well

OpenSauce04 commented on src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/SwitchSettingViewHolder.kt:
```diff
@@ -45,18 +59,36 @@ class SwitchSettingViewHolder(val binding: ListItemSettingSwitchBinding, adapter
+                return false
+            } else {
+                return adapter.onLongClick(setting.setting!!, bindingAdapterPosition)
+            }
```
Invert this if statement and remove the unnecessary `else`.

OpenSauce04 commented on src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/SwitchSettingViewHolder.kt:
```diff
@@ -45,18 +59,36 @@ class SwitchSettingViewHolder(val binding: ListItemSettingSwitchBinding, adapter
+                adapter.onForceMaximumGpuClocksDisabled()
+            } else {
+                binding.switchWidget.toggle()
+            }
```
Invert this if statement.

OpenSauce04 commented on src/android/app/src/main/java/io/github/lime3ds/android/features/settings/ui/viewholder/SwitchSettingViewHolder.kt:
```diff
@@ -33,10 +37,20 @@ class SwitchSettingViewHolder(val binding: ListItemSettingSwitchBinding, adapter
+            } else {
+                binding.textSettingName.alpha = 1f
+                binding.textSettingDescription.alpha = 1f
+            }
```
Set a variable to either 0.5 or 1 depending on the condition, then set `binding.textSettingName.alpha` and `binding.textSettingDescription.alpha` to the value of that variable.

Commit:
Replace all instances of `Clocks` with `Clock Speed`

Commit:
Small Correction

Commit:
Simplify code on `SwitchSettingViewHolder.kt`

Commit:
Invert `if statement` and remove `else`

on `SwitchSettingViewHolder.kt`

Commit:
Invert another if statement

on `SwitchSettingViewHolder.kt`

Commit:
Use a variable to set the alpha values

OpenSauce04 committed:
Fix crash when loading into a game with Adreno Turbo enabled

OpenSauce04 committed:
Adreno Turbo setting is now hidden instead of disabled when unavailable

This saves unnecessary code complications which would otherwise be present with the previous implementation

It also removes unnecessary settings for people who will never be able to use them because of their hardware not supporting the feature

OpenSauce04 committed:
Updated Adreno Turbo setting strings and keys

OpenSauce04 committed:
Updated Adreno GPU Boost setting title and description
